run/jenkins.py

```diff
@@ -0,0 +1,282 @@
+    args = {
+        'prod_run': bool(os.environ.get('PROD_RUN', False)),
+        'prod_wet_run': bool(os.environ.get('PROD_WET_RUN', False)),
+        'SHA': os.environ.get('WPT_SHA'),
+        'sauce_from_metadata': bool(os.environ.get('SAUCE_FROM_METADATA')),
+        'sauce_key': os.environ.get('SAUCE_KEY', ''),
+        'sauce_user': os.environ.get('SAUCE_USER', ''),
+        'jenkins_job_name': os.environ.get('JOB_NAME'),
+        'wpt_path': os.environ.get('WPT_PATH'),
+        'wptd_path': '/wptdashboard',
+        'output_path': '/wptdout',
+        'run_path': os.environ.get('RUN_PATH', ''),
+    }
+
+    # Arg validation
+    assert args['wpt_path'], '`WPT_PATH` env var required.'
+    assert not args['wpt_path'].endswith('/'), '`WPT_PATH` cannot end with /.'
+
+    if args['prod_run'] or args['prod_wet_run']:
+        print 'Fetching upload_secret from GCE Metadata...'
+
+        url = '%s/project/attributes/upload_secret' % METADATA_URL
+        res = requests.get(url, headers={'Metadata-Flavor': 'Google'})
+        args['upload_secret'] = res.text
+        assert len(args['upload_secret']) == 64, (
+            'Metadata `upload_secret` must exist for prod runs.')
+
+    platform_id, platform = get_and_validate_platform(args['wptd_path'])
+
+    PROD_HOST = 'https://wptdashboard.appspot.com'
+    GS_RESULTS_BUCKET = 'wptd'
+    LOCAL_LOG_FILEPATH = '%s/wptd-testrun.log' % args['output_path']
+    LOCAL_REPORT_FILEPATH = "%s/wptd-%s-%s-report.log" % (
+        args['output_path'], args['SHA'], platform_id
+    )
+    SUMMARY_PATH = '%s/%s-summary.json.gz' % (args['SHA'], platform_id)
+    LOCAL_SUMMARY_GZ_FILEPATH = "%s/%s" % (args['output_path'], SUMMARY_PATH)
+    GS_RESULTS_FILEPATH_BASE = "%s/%s/%s" % (
+        args['output_path'], args['SHA'], platform_id
+    )
+    GS_HTTP_RESULTS_URL = 'https://storage.googleapis.com/%s/%s' % (
+        GS_RESULTS_BUCKET, SUMMARY_PATH
+    )
+
+    SUMMARY_FILENAME = '%s-%s-summary.json.gz' % (args['SHA'], platform_id)
+    SUMMARY_HTTP_URL = 'https://storage.googleapis.com/%s/%s' % (
+        GS_RESULTS_BUCKET, SUMMARY_FILENAME
+    )
+
+    if platform.get('sauce'):
+        if args['sauce_from_metadata']:
+            print 'Fetching Sauce creds from GCE metadata...'
+
+            for key in ('sauce_user', 'sauce_key'):
+                url = '%s/project/attributes/%s' % (METADATA_URL, key)
+                res = requests.get(url, headers={'Metadata-Flavor': 'Google'})
+                args[key] = res.text
+                assert args[key], 'Metadata key %s is empty.' % key
+
+        assert args['sauce_key'], 'SAUCE_KEY env var required'
+        assert args['sauce_user'], 'SAUCE_USER env var required'
+        SAUCE_TUNNEL_ID = '%s_%s' % (platform_id, int(time.time()))
+
+    assert len(args['SHA']) == 10, 'SHA must a WPT SHA[:10]'
+
+    # Hack because Sauce expects a different name
+    # Maybe just change it in browsers.json?
```

Comment:
IMHO browsers.json should have a schema that includes an optional `sauce_name`. Code can fallback on `browser_name` when it's not set.

Reply:
That would work. I've been keeping the patch around in wpt.fyi because this problem is an actual bug in wptrunner: https://github.com/w3c/web-platform-tests/issues/6852

I've been meaning to write a fix for that at some point.

Reply:
Ah. So we should be able to eliminate the need for this soon. Great. Can you please link to the wpt bug?

Reply:
Oops I totally goofed, this is not the hack I was thinking of. Your suggestion is good. I'll make a bug for that.

Reply:
#201